### PR TITLE
[ISSUE#7519] Optimize configuration history deletion logic

### PR DESCRIPTION
**Pull request by hujun-w-2**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

for #7519

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by realJackSun — DISMISSED** [review]

**Comment by realJackSun** [comment]
@TsingLiang @shiyiyue1102
 This PR looks good for me, I approved. 
Could you plz review it.

**Comment by realJackSun** [comment]
@li-xiao-shuang How do you see this?

File: config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java [hunk 1]
```diff
@@ -631,11 +632,20 @@ public boolean batchRemoveAggr(final String dataId, final String group, final St
...
+        ExternalStoragePaginationHelperImpl<Object> paginationHelper = (ExternalStoragePaginationHelperImpl<Object>) createPaginationHelper();
+        int count;
         try {
-            helper.updateLimit(sql, new Object[] {startTime, limitSize});
+            count = paginationHelper.updateLimitWithResponse(sql, new Object[] {startTime, limitSize});
+            while (count > 0) {
+                try {
+                    TimeUnit.SECONDS.sleep(1);
+                } catch (InterruptedException e) {
+                    LogUtil.FATAL_LOG.error("[db-error] " + e, e);
```
Comment (li-xiao-shuang) [inline config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java:643]: 这里的日志可以在明确一下  此处应该不是db 的error

File: config/src/main/java/com/alibaba/nacos/config/server/service/dump/DumpService.java [hunk 2]
```diff
@@ -182,19 +182,9 @@ protected void dumpOperate(DumpProcessor processor, DumpAllProcessor dumpAllProc
                 if (canExecute()) {
                     try {
                         Timestamp startTime = getBeforeStamp(TimeUtils.getCurrentTime(), 24 * getRetentionDays());
-                        int totalCount = persistService.findConfigHistoryCountByTime(startTime);
```
Comment (li-xiao-shuang) [inline config/src/main/java/com/alibaba/nacos/config/server/service/dump/DumpService.java:185]: findConfigHistoryCountByTime() 这个方法我认为可以用 @Deprecated 标记下，避免以后的错误使用

File: config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java [hunk 3]
```diff
@@ -631,11 +632,20 @@ public boolean batchRemoveAggr(final String dataId, final String group, final St
     @Override
     public void removeConfigHistory(final Timestamp startTime, final int limitSize) {
         String sql = "DELETE FROM his_config_info WHERE gmt_modified < ? LIMIT ?";
-        PaginationHelper<ConfigInfo> helper = createPaginationHelper();
+        ExternalStoragePaginationHelperImpl<Object> paginationHelper = (ExternalStoragePaginationHelperImpl<Object>) createPaginationHelper();
```
Comment (li-xiao-shuang) [inline config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java:635]: ExternalStoragePaginationHelperImpl<Object> 是不是可以替换为 ExternalStoragePaginationHelperImpl< ConfigInfo > ？

**Review by li-xiao-shuang — APPROVED** [review]

**Comment by li-xiao-shuang** [comment]
> @li-xiao-shuang How do you see this?

look good

**Review by realJackSun — APPROVED** [review]

**Comment by ADM130066** [comment]
> Please do not create a Pull Request without creating an issue first.
> 
> ## What is the purpose of the change
> 
> for #7519
> 
> ## Brief changelog
> 
> XX
> 
> ## Verifying this change
> 
> XXXX
> 
> Follow this checklist to help us incorporate your contribution quickly and easily:
> 
> * [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
> * [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
> * [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
> * [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
> * [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
> 
>

**Pull request merged**